### ISSUE TITLE
union(::Set, itr...), intersect(::Set, itr...), and setdiff(::Set, itr) should return Set and not Array{Int,1}

### ISSUE THREAD

**Issue by KlausC** [JuliaLang/julia#23137]
This issue refers to
https://discourse.julialang.org/t/union-and-intersect-with-unexpected-return-types/5222
```
julia> union(Set(Int16[1,2,3]), IntSet([1,42]))
4-element Array{Int64,1}:
  2
  3
  1
 42
julia> intersect(Set(Int16[1,2,3]), IntSet([1,42]))
1-element Array{Int64,1}:
 1
julia> setdiff(Set(Int16[1,2,3]), IntSet(2:4))
1-element Array{Int64,1}:
 1
```
The method dispatches to the array-version for the example, but schould use the set-version:
```
julia> @which union(Set([1]), IntSet())
union(vs...) in Base at array.jl:1870
julia> @which union(Set([1]), Set{Int}())
union(s::Set, sets::Set...) in Base at set.jl:55
```
propose to change `set.jl`:
`function union(s::Set, sets...)`
`function intersect(s::Set, sets...)`
`function setdiff(s::Set, xs)`

**Comment by nalimilan** [comment]
It would be good to define a general rule for the choice of the return type of `union`, `intersect` and `setdiff`. A simple rule is to use the type of the first argument. A possible better one (which could be implemented only later) would be to use a promotion mechanism, maybe the same one as for concatenation (cf. https://github.com/JuliaLang/julia/issues/18472).

**Comment by rfourquet** [comment]
Closed by #23139.
